File: src/main/java/cs/android/viewbase/CSOnMenuItem.java
package cs.android.viewbase;

import android.view.MenuItem;

import cs.java.lang.CSValue;

import static cs.java.lang.CSLang.NO;
import static cs.java.lang.CSLang.no;

public class CSOnMenuItem {
    private final CSValue<Boolean> _consumed;
    private final MenuItem _item;

    public CSOnMenuItem(MenuItem item) {
        _item = item;
        _consumed = new CSValue<>(false);
    }

    public boolean consume(int id) {
        if (_consumed.get()) return NO;
        _consumed.set(_item.getItemId() == id);
        return _consumed.get();
    }

    public boolean consume(MenuItem item) {
        if (no(item)) return NO;
        return consume(item.getItemId());
    }

    public boolean consumed() {
        return _consumed.get();
    }

    public boolean isCheckable() {
        return _item.isCheckable();
    }

    public boolean isChecked() {
        return _item.isChecked();
    }

    public void checked(boolean checked) {
        _item.setChecked(checked);
    }

}

File: library/src/main/java/renetik/android/viewbase/menu/CSOnMenuItem.kt
package renetik.android.viewbase.menu

import android.view.MenuItem
import renetik.android.lang.CSLang.NO
import renetik.android.lang.CSLang.set

class CSOnMenuItem(private val item: MenuItem) {
    var isConsumed = false

    val isCheckable
        get() = item.isCheckable

    var isChecked
        get() = item.isChecked
        set(value) {
            item.isChecked = value
        }


    fun consume(item: CSMenuItem): Boolean {
        if (isConsumed) return NO
        if (set(item.id)) isConsumed = this.item.itemId == item.id
        return isConsumed
    }

    fun consume(id: Int): Boolean {
        if (isConsumed) return NO
        isConsumed = item.itemId == id
        return isConsumed
    }

}